Source/DataSources/StaticGroundPolylinePerMaterialBatch.js

```diff
@@ -0,0 +1,365 @@
+                    if (!DistanceDisplayCondition.equals(distanceDisplayCondition, attributes._lastDistanceDisplayCondition)) {
+                        attributes._lastDistanceDisplayCondition = DistanceDisplayCondition.clone(distanceDisplayCondition, attributes._lastDistanceDisplayCondition);
+                        attributes.distanceDisplayCondition = DistanceDisplayConditionGeometryInstanceAttribute.toValue(distanceDisplayCondition, attributes.distanceDisplayCondition);
+                    }
+                }
+            }
+
+            this.updateShows(primitive);
+        } else if (defined(primitive) && !primitive.ready) {
+            isUpdated = false;
+        }
+        return isUpdated;
+    };
+
+    Batch.prototype.updateShows = function(primitive) {
+        var showsUpdated = this.showsUpdated.values;
+        var length = showsUpdated.length;
+        for (var i = 0; i < length; i++) {
+            var updater = showsUpdated[i];
+            var entity = updater.entity;
+            var instance = this.geometry.get(updater.id);
+
+            var attributes = this.attributes.get(instance.id.id);
+            if (!defined(attributes)) {
+                attributes = primitive.getGeometryInstanceAttributes(instance.id);
+                this.attributes.set(instance.id.id, attributes);
+            }
+
+            var show = entity.isShowing;
+            var currentShow = attributes.show[0] === 1;
+            if (show !== currentShow) {
+                attributes.show = ShowGeometryInstanceAttribute.toValue(show, attributes.show);
+            }
+        }
+        this.showsUpdated.removeAll();
+    };
+
+    Batch.prototype.contains = function(updater) {
+        return this.updaters.contains(updater.id);
+    };
+
+    Batch.prototype.getBoundingSphere = function(updater, result) {
+        var primitive = this.primitive;
+        if (!primitive.ready) {
+            return BoundingSphereState.PENDING;
+        }
+        var attributes = primitive.getGeometryInstanceAttributes(updater.entity);
+        if (!defined(attributes) || !defined(attributes.boundingSphere) ||
+            (defined(attributes.show) && attributes.show[0] === 0)) {
+            return BoundingSphereState.FAILED;
+        }
+        attributes.boundingSphere.clone(result);
+        return BoundingSphereState.DONE;
+    };
+
+    Batch.prototype.destroy = function() {
+        var primitive = this.primitive;
+        var orderedGroundPrimitives = this.orderedGroundPrimitives;
+        if (defined(primitive)) {
+            orderedGroundPrimitives.remove(primitive);
+        }
+        var oldPrimitive = this.oldPrimitive;
+        if (defined(oldPrimitive)) {
+            orderedGroundPrimitives.remove(oldPrimitive);
+        }
+        this.removeMaterialSubscription();
+    };
+
+    /**
+     * @private
+     */
+    function StaticGroundPolylinePerMaterialBatch(orderedGroundPrimitives) {
```

Comment:
This is mostly copied from `StaticGroundGeometryPerMaterialBatch`, it just allows a lot more flexibility in batching and needs its own primitive-type.